message/event/uploadEvent.go

```diff
@@ -0,0 +1,42 @@
+package event
+
+import (
+	"encoding/json"
+	"net/url"
+	"strings"
+)
+
+type UploadEvent struct {
+	Time  int64  `json:"time"`
+	S3URL string `json:"s3URL"`
+}
+
+func (event *UploadEvent) ConvertToEventDetails() (*EventDetails, error) {
+	s3url, err := url.Parse(event.S3URL)
+	if err != nil {
+		return nil, err
+	}
+	_, err = url.ParseRequestURI(s3url.RequestURI())
```

Comment:
it should, it does the same thing internally:
https://golang.org/src/net/url/url.go?s=11553:11592#L408